system/doc/efficiency_guide/maps.xml

```diff
@@ -0,0 +1,647 @@
+  finally the functions in the <seeerl
+  marker="stdlib:maps">maps</seeerl> module.</p>
+
+  <p>Terminology used in this chapter:</p>
+  <list type="bulleted">
+    <item>A map with at most 32 elements will informally be called a
+    <em>small map</em>.</item>
+    <item>A map with more than 32 elements will informally be called a
+    <em>large map</em>.</item>
+  </list>
+
+  <section>
+    <title>Maps or Records?</title>
+    <p>If the advice in this chapter is followed, the performance of
+    records compared to using small maps instead of records is
+    expected to be similar. Therefore, the choice between records and
+    maps should be based on the desired properties of the data
+    structure and not performance.</p>
+
+    <p>The advantages of records compared to maps are:</p>
+
+    <list type="bulleted">
+      <item>If the name of a record field is misspelled, there will be
+      a compilation error. If a map key is misspelled, the compiler
+      will give no warning and program will fail in some way when it
+      is run.</item>
+
+      <item>Records will use slightly less memory than maps, and
+      performance is expected to be <em>slightly</em> better than maps
+      in most circumstances.</item>
+    </list>
+
+    <p>The disadvantage of records compared to maps is that if a new
+    field is added to a record, all code that uses that record must be
+    recompiled. Because of that, it is recommended to only use records
+    within a unit of code that can easily be recompiled all at once, for
+    example within a single application or single module.</p>
+  </section>
+
+  <section>
+    <title>Using Maps as an Alternative to Records</title>
+
+    <list type="bulleted">
+      <item><p>Use the map syntax instead of the functions in
+      the <seeerl marker="stdlib:maps">maps</seeerl> module.</p></item>
+
+      <item><p>Avoid having more than 32 elements in the
+      map. As soon as there are more than 32 elements in the map,
+      it will require more memory and keys can no longer be shared
+      with other instances of the map.</p></item>
+
+      <item><p>When creating a new map, always create it with all keys
+      that will ever be used. To maximize sharing of keys (thus
+      minimizing memory use), create a single function that constructs
+      the map using the map syntax and always use it.</p></item>
+
+      <item><p>Always update the map using the
+      <c>:=</c> operator (that is, requiring that an element with
+      that key already exists).  The <c>:=</c> operator is slightly
+      more efficient, and it helps catching mispellings of
+      keys.</p></item>
+
+      <item><p>Whenever possible, match multiple map elements at
+      once.</p></item>
+
+      <item><p>Whenever possible, update multiple map elements at
+      once.</p></item>
+
+      <item><p>Avoid default values and the <seemfa
+      marker="stdlib:maps#get/3">maps:get/3</seemfa>
+      function.</p></item>
```

Comment:
This is covered later. I said in the introduction that the those first sections would be brief, but that we would dig deeper in later sections.

I think that the items should be separate. The first item says to avoid default values, the other suggests **one** way to do so.